mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -2690,49 +2747,65 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
-          isOutputPad = false;
-          isGemmDim1Pad = false;
+          isOutputPad = true;
+          isGemmDim1Pad = true;
+          gemmDim1TargetName = b.getStringAttr(gemmMPad_name);
+          // fwd, k
           paddingOutputShape[1] = paddingOutputShape[1] + gemmMExtra;
+          sourceGemmDim1Attr.push_back(
+              b.getNamedAttr("transformation", b.getStringAttr("Pad")));
+          sourceGemmDim1Attr.push_back(b.getNamedAttr(
+              "parameters", b.getArrayAttr({b.getI32IntegerAttr(0),
+                                            b.getI32IntegerAttr(gemmMExtra)})));
+
+          targetGemmDim1Attr.push_back(b.getNamedAttr(
+              "names", b.getArrayAttr({b.getStringAttr(gemmMPad_name)})));
+          outputOobCheckDims.insert(nameToDims["ko"]);
         } else if (arg2TargetLayoutName2 == "gemmM") {
-          isOutputPad = false;
-          isGemmDim2Pad = false;
+          isOutputPad = true;
+          isGemmDim2Pad = true;
+          gemmDim2TargetName = b.getStringAttr(gemmMPad_name);
+          // wrw ,k
           paddingOutputShape[2] = paddingOutputShape[2] + gemmMExtra;
+          sourceGemmDim2Attr.push_back(
+              b.getNamedAttr("transformation", b.getStringAttr("Pad")));
+          sourceGemmDim2Attr.push_back(b.getNamedAttr(
+              "parameters", b.getArrayAttr({b.getI32IntegerAttr(0),
+                                            b.getI32IntegerAttr(gemmMExtra)})));
+
+          targetGemmDim2Attr.push_back(b.getNamedAttr(
+              "names", b.getArrayAttr({b.getStringAttr(gemmMPad_name)})));
+          outputOobCheckDims.insert(nameToDims["ko"]);
         }
       }
 
-      if (gemmNExtra > 0) {
-        if (arg2TargetLayoutName1 == "gemmN") {
-          isOutputPad = false;
-          isGemmDim1Pad = false;
-          paddingOutputShape[1] = paddingOutputShape[1] + gemmNExtra;
-        } else if (arg2TargetLayoutName2 == "gemmN") {
-          isOutputPad = false;
-          isGemmDim2Pad = false;
+      if (outputCheckPadGemmN) {
+        if (arg2TargetLayoutName2 == "gemmN") {
+          // fwd dim 2,nhw
+          isOutputPad = true;
+          isGemmDim2Pad = true;
+          gemmDim2TargetName = b.getStringAttr(gemmNPad_name);
           paddingOutputShape[2] = paddingOutputShape[2] + gemmNExtra;
+          sourceGemmDim2Attr.push_back(
+              b.getNamedAttr("transformation", b.getStringAttr("Pad")));
+          sourceGemmDim2Attr.push_back(b.getNamedAttr(
+              "parameters", b.getArrayAttr({b.getI32IntegerAttr(0),
+                                            b.getI32IntegerAttr(gemmNExtra)})));
+
+          targetGemmDim2Attr.push_back(b.getNamedAttr(
+              "names", b.getArrayAttr({b.getStringAttr(gemmNPad_name)})));
+          // FIXME: to set dim in merge transormation to oob store,
+          // set only top dim or you will get zero values
+          outputOobCheckDims.insert(nameToDims["no"]);
```

Comment:
please add comments here on the rationale assigning dimensions to OOB check dimensions, like what you did above for filter.

Reply:
comment was not added.

Reply:
line 2793